exporter/otlp/lib/opentelemetry/exporter/otlp/exporter.rb

```diff
@@ -117,6 +118,11 @@ def send_bytes(bytes) # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticCompl
             # TODO: enable gzip when https://github.com/open-telemetry/opentelemetry-collector/issues/1344 is fixed.
             # request.add_field('Content-Encoding', 'gzip')
 
+            remaining_timeout = OpenTelemetry::SDK::Internal.maybe_timeout(timeout, start_time)
+            return FAILURE if remaining_timeout.zero?
+
+            @http.open_timeout = remaining_timeout
+            @http.read_timeout = remaining_timeout
```

Comment:
We would like to set `@http.write_timeout` here too, but that isn't available in Ruby 2.5. Shopify's internal implementation has a version check to allow setting that for Ruby versions > 2.5. Do we want to do the same thing here?

Reply:
I think it makes sense to add it if we're diligent enough to drop the version check when 2.5 hits EoL.

Reply:
Added write_timeout.